### DISCUSSION TITLE
How to access data when there are errors?

### DISCUSSION THREAD

**Discussion by jim-moody** [TanStack/query#3365] [General] [answered]
In some cases, for example a list of documents, there is an error with a few of the documents but the rest of the documents are returned successfully.

However, react-query does not return any `data` when there are errors. I don't think this follows the graphql standard because the whole point of exposing a field for `data` and a field for `errors` is so that a client can deal with those two things independently.

**Comment by TkDodo** [comment, marked as answer]
react-query is promise based and agnostic to any data fetching technique. A promise can be resolved, in which case you'll get `data`, or rejected, in which case you get an `error`, and potential retries.

if you have one request that somehow returns data for some parts and errors for other parts, like graphql allows you to do (technically, everything graphql does is a resolved promise because everything is 200 - OK on response level 🙈), then the best thing you can do is to resolve the promise, keep everything in `data` and handle it from there.

So no, it does not follow graphql standards because it's not a graphql specific library. If you need graphql specific features, I'm afraid that is something that react-query cannot offer.

**Reply by jim-moody** [reply]
Ok that makes sense. I am using react-query with graphql-code-generator so all my queries are directly tied to graphql which is why I am thinking in such a graphql-specific way but of course there are other standards, thanks!
